src/main/kotlin/org/rust/ide/formatter/RustAlignmentStrategy.kt

```diff
@@ -0,0 +1,42 @@
+package org.rust.ide.formatter
+
+import com.intellij.formatting.Alignment
+import com.intellij.lang.ASTNode
+
+/**
+ * Mimics [com.intellij.formatting.alignment.AlignmentStrategy], but offers more flexible API
+ */
+interface RustAlignmentStrategy {
```

Comment:
What do you think about such API?
```kotlin
fun RustFmtBlock.getAlignmentStrategy(): RustAlignmentStrategy = when (node.elementType) {
    TUPLE_EXPR, TUPLE_TYPE, ARG_LIST, ENUM_TUPLE_ARGS ->
        RustAlignmentStrategy.shared()
            .cond { c, p -> !c.isBlockDelim(p) } 
            .cfg(...) // enable configuration

    in PARAMS_LIKE ->
        RustAlignmentStrategy.lazy { this.getUserData(PARAMETERS_ALIGNMENT) } // actually this can be implemented as a real lazy
            .cond { c, p -> !c.isBlockDelim(p) }
            .cfg(...)

    in FN_DECLS ->
        RustAlignmentStrategy.lazy { this.getUserData(PARAMETERS_ALIGNMENT) }
            .filter(TokenSet.create(RET_TYPE, WHERE_CLAUSE))
            .cfg(...)

    PAT_ENUM ->
        RustAlignmentStrategy.shared()
            .cond { c, p -> this.getUserData(INDENT_MET_LBRACE) == true && !c.isBlockDelim(p) }

    else -> RustAlignmentStrategy.NULL_STRATEGY
}
```

The problem with plain functions is that there is no clean way to store alignment object